### DISCUSSION TITLE
run MTEX functions on GPU?

### DISCUSSION THREAD

**Discussion by Remsdel** [mtex-toolbox/mtex#1421] [Ask Anything]
Does anybody has experience running some MTEX functions on GPU (if this is even possible)?
I have a computer with a lot of GPU processing power and would like to make use of it in the hope of getting faster MTEX data processing.
MATLAB has the gpuArray function that store arrays on the GPU and allows some functions to run on the GPU.
It is however not straightforward to figure out on which MTEX variables one could apply gpuArray and if MTEX functions are even gpuArray-enabled.
Thanks for the feedback,
Cordially,
Rémi Delville

**Comment by ralfHielscher** [comment]
I think the first question is, which are the time consuming tasks you want to speed up?
E.g. if you have large EBSD maps, it might the grain reconstruction. So the answer depends very much on the specific functions you would like to speed up.

Ralf.

**Comment by Remsdel** [comment]
Yes indeed, grain reconstruction for large maps
________________________________

**Comment by lisakemp** [comment]
Hello,

Lisa from MathWorks here.

As Remi considers which functions are the targets for speedup, per Ralf’s question, this doc page that contains pointers to the lists of GPU-enabled functions for MATLAB and toolboxes might help:

https://www.mathworks.com/help/parallel-computing/run-matlab-functions-on-a-gpu.html

Lisa

**Comment by Remsdel** [comment]
Thanks Lisa,
I have of course looked at this page you mentioned.

However, it would take a lot of detective work to figure all the functions behind a MTEX function, such as calcGrains (one of the most CPU-consuming) that could potentially benefit from GPU computation. It also requires to figure out which of the ebsd embedded data (ebsd.xxx) on which to apply gpuArray. It seems to me far from trivial and I was hoping that somebody had done the exercise. That could potentially lead to much faster processing for large EBSD maps. I hope there are more people interested in this.

Cordially,

Rémi

**Comment by lisakemp** [comment]
Remi – That all make sense.

Yes, it would be work to figure out all the dependencies (i.e. subfunctions included in the hierarchy of a function call), and, even then, the speed-up implications/opportunities given those functions’ GPU-enabled state.

Regardless, perhaps the details of which MATLAB functions are GPU-enabled will be of use along the way.

Interesting questions.

Lisa

**Reply by kilir** [reply]
Have you actually tested if your gpu setup is actually faster for double precision type data?
Rüdiger
